spec/reflexes/products_reflex_spec.rb

```diff
@@ -53,7 +54,33 @@
         run_reflex(:bulk_update, params:)
         product_a.reload
       }.to change{ product_a.name }.to("Pommes")
-       .and change{ product_a.sku }.to("POM-01")
+        .and change{ product_a.sku }.to("POM-01")
+    end
+
+    it "saves valid changes to nested variants" do
+      params = {
+        # '[products][][name]'
+        # '[products][][variants_attributes][][display_name]'
+        "products" => [
+          {
+            "id" => product_a.id.to_s,
+            "name" => "Pommes",
+            "variants_attributes" => [
+              {
+                "id" => variant_a1.id.to_s,
+                "display_name" => "Large box",
+              }
```

Comment:
Reading the code I wasn't sure if the resulting params would put "id" and "display_name" in their own hashes as two elements of the array. If so then it would probably be better to replicate that here.